### DISCUSSION TITLE
Why can't I add a string to an array property that does not exist?

### DISCUSSION THREAD

**Discussion by technicallyfeasible** [dynamoose/dynamoose#1177] [Q&A]
I'm pretty sure this worked in v1. Now it throws an error "The provided expression refers to an attribute that does not exist in the item".

This was in the schema:
```javascript
messages: {
  type: [String]
}
```

and we would insert the first message like so:
```javascript
await CommandStatus.update({
  ...
}, {
  $ADD: {
    messages: [JSON.stringify(message)],
  },
});
```

**Comment by technicallyfeasible** [comment]
@fishcharlie

**Comment by fishcharlie** [comment]
So in v2 your schema needs to be changed to:

```js
messages: {
  type: Set,
  schema: [String]
}
```

That will basically set `messages` to be a set of strings in v2.

But if you look at the following line, in v1 this isn't doing anything special:

https://github.com/dynamoose/dynamoose/blob/9e5fb1605e99da2d5550cfa896d224c09aaa3cc8/lib/Model.js#L947

It's just making a standard add call to the update expression. Which v2 does as well:

https://github.com/dynamoose/dynamoose/blob/40443acc6f7a07681592f294d3893902c3652e34/lib/Model/index.ts#L776

So I'm really not sure what is going on here other than the schema change I mentioned above.

If that change above doesn't work, I'd be curious what the difference is in the AWS calls that are made. I'm not sure what v1 is doing that v2 isn't. So you might have to enable logging on both platforms and post the requests that are made to AWS here. In v2 you should be able to do that by adding `dynamoose.logger.providers.set([console]);` to your code. In v1 you should be able to do it by setting the `DEBUG` environment variable to `dynamoose*`.
